src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation of the Function Logger
+"""
+from typing import Any, Callable
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from deepsparse.loggers.helpers import match, possibly_extract_value
+
+
+__all__ = ["FunctionLogger"]
+
+
+class FunctionLogger(BaseLogger):
+    """
+    DeepSparse logger that applies functions to raw,
+    logged values (collected in the log() method)
+    according to FunctionLogger's attributes
+
+    :param logger: A child DeepSparse Logger object
+    :param identifier: The string that needs to match the
+        `identifier` (argument to the log() method),
+        so that the FunctionLogger applies
+        the metric function and logs the data
+    :param function_name: Name of the metric function
+    :param function: The metric function to be applied
+    :param frequency: The frequency with which the metric
+        name is to be applied
+    """
+
+    def __init__(
+        self,
+        logger: BaseLogger,
+        identifier: str,
+        function_name: Any,
```

Comment:
let's add a default here potentially None / extract the name from the function reference